velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -105,72 +105,110 @@ TEST_F(DecimalArithmeticTest, add) {
-      resultLongFlat, "minus(c0, c1)", {longFlatA, shortFlatA});
+      makeLongDecimalFlatVector({-900, -1800}, DECIMAL(20, 3)),
+      "minus(c0, c1)",
+      {longFlatA, shortFlatA});
 
   // Subtract long and long, returning long.
-  auto longFlatB = makeLongDecimalFlatVector({100, 200}, DECIMAL(19, 2));
-  resultLongFlat = makeLongDecimalFlatVector({-900, -1800}, DECIMAL(21, 3));
   testDecimalExpr<TypeKind::LONG_DECIMAL>(
-      resultLongFlat, "c0 - c1", {longFlatA, longFlatB});
+      makeLongDecimalFlatVector({-900, -1800}, DECIMAL(21, 3)),
+      "c0 - c1",
+      {longFlatA, makeLongDecimalFlatVector({100, 200}, DECIMAL(19, 2))});
 
   // Subtract short and short, returning short.
-  shortFlatA = makeShortDecimalFlatVector({1000, 2000}, DECIMAL(10, 3));
-  shortFlatB = makeShortDecimalFlatVector({500, 1000}, DECIMAL(10, 3));
-  auto resultShortFlat =
-      makeShortDecimalFlatVector({500, 1000}, DECIMAL(11, 3));
   testDecimalExpr<TypeKind::SHORT_DECIMAL>(
-      resultShortFlat, "minus(c0, c1)", {shortFlatA, shortFlatB});
-  auto resultConstantFlat =
-      makeShortDecimalFlatVector({0, -1000}, DECIMAL(11, 3));
-
+      makeShortDecimalFlatVector({500, 1000}, DECIMAL(11, 3)),
+      "minus(c0, c1)",
+      {makeShortDecimalFlatVector({1000, 2000}, DECIMAL(10, 3)),
+       makeShortDecimalFlatVector({500, 1000}, DECIMAL(10, 3))});
   // Constant and Flat arguments.
+  shortFlatA = makeShortDecimalFlatVector({1000, 2000}, DECIMAL(10, 3));
   testDecimalExpr<TypeKind::SHORT_DECIMAL>(
-      resultConstantFlat, "minus(1.00, c0)", {shortFlatA});
-  resultConstantFlat = makeShortDecimalFlatVector({0, 1000}, DECIMAL(11, 3));
+      makeShortDecimalFlatVector({0, -1000}, DECIMAL(11, 3)),
+      "minus(1.00, c0)",
+      {shortFlatA});
 
   // Flat and Constant arguments.
   testDecimalExpr<TypeKind::SHORT_DECIMAL>(
-      resultConstantFlat, "minus(c0, 1.00)", {shortFlatA});
+      makeShortDecimalFlatVector({0, 1000}, DECIMAL(11, 3)),
+      "minus(c0, 1.00)",
+      {shortFlatA});
 
   // Input with NULLs.
-  auto shortWithNullsA = makeNullableShortDecimalFlatVector(
-      {3, 6, std::nullopt, 6, std::nullopt}, DECIMAL(10, 3));
-  auto shortWithNullsB = makeNullableShortDecimalFlatVector(
-      {1, 2, 5, std::nullopt, std::nullopt}, DECIMAL(10, 3));
-  auto expectedShortWithNulls = makeNullableShortDecimalFlatVector(
-      {2, 4, std::nullopt, std::nullopt, std::nullopt}, DECIMAL(11, 3));
   testDecimalExpr<TypeKind::SHORT_DECIMAL>(
-      expectedShortWithNulls,
+      makeNullableShortDecimalFlatVector(
+          {2, 4, std::nullopt, std::nullopt, std::nullopt}, DECIMAL(11, 3)),
       "minus(c0, c1)",
-      {shortWithNullsA, shortWithNullsB});
+      {makeNullableShortDecimalFlatVector(
+           {3, 6, std::nullopt, 6, std::nullopt}, DECIMAL(10, 3)),
+       makeNullableShortDecimalFlatVector(
+           {1, 2, 5, std::nullopt, std::nullopt}, DECIMAL(10, 3))});
+}
+
+TEST_F(DecimalArithmeticTest, decimalMultiplyTest) {
```

Comment:
nit: would it be possible to have tests around numeric limits of these decimals (?) .